src/main/java/uk/ac/cam/cl/dtg/isaac/api/AssignmentFacade.java

```diff
@@ -233,6 +233,63 @@ public Response getAssignments(@Context final HttpServletRequest request,
         }
     }
 
+    /**
+     * Allows the user to fetch a single assignment that they own (or are a group manager of). Returned assignment
+     * will contain gameboard and question information.
+     *
+     * @param request - so that we can identify the current user.
+     * @param assignmentId - id of assignment to fetch
+     * @return AssignmentDTO containing extra information about the gameboard and questions
+     */
+    @GET
+    @Path("/assign/{assignmentId}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @GZIP
+    @Operation(summary = "Fetch an assignment object populated with gameboard and quesiton information, if the user "
+                        + "owns the group that the assignment is set to.")
+    public Response getSingleAssigned(@Context final HttpServletRequest request,
+                                @PathParam("assignmentId") final Long assignmentId) {
+        if (null == assignmentId) {
+            return new SegueErrorResponse(Status.BAD_REQUEST, "Please specify an assignment id to search for.").toResponse();
+        }
+        try {
+            RegisteredUserDTO currentlyLoggedInUser = userManager.getCurrentRegisteredUser(request);
+
+            AssignmentDTO assignment = this.assignmentManager.getAssignmentById(assignmentId);
+            if (null == assignment) {
+                return new SegueErrorResponse(Status.NOT_FOUND, String.format("Assignment with id %d not found.", assignmentId)).toResponse();
+            }
+            UserGroupDTO group = this.groupManager.getGroupById(assignment.getGroupId());
+            if (null == group) {
+                return new SegueErrorResponse(Status.INTERNAL_SERVER_ERROR, "Error while locating the specified assignment.").toResponse();
+            }
+
+            if (!GroupManager.isOwnerOrAdditionalManager(group, currentlyLoggedInUser.getId())
+                    && !isUserAnAdmin(userManager, currentlyLoggedInUser)) {
+                return new SegueErrorResponse(Status.FORBIDDEN, "You are not the owner or manager of this assignment.").toResponse();
+            }
+
+            // In order to get all the information about parts and pass marks, which is needed for Assignment Progress,
+            // we need to use the method which augments a gameboard with user attempt information.
+            // But we don't _want_ the attempt information itself for real, so we won't load it from the database:
+            Map<String, Map<String, List<QuestionValidationResponse>>> fakeQuestionAttemptMap = new HashMap<>();
+
+            assignment.setGameboard(this.gameManager.getGameboard(assignment.getGameboardId(), currentlyLoggedInUser, fakeQuestionAttemptMap));
+
+            this.getLogManager().logEvent(currentlyLoggedInUser, request, IsaacServerLogType.VIEW_ASSIGNMENT,
+                    ImmutableMap.of("assignmentId", assignmentId));
```

Comment:
I don't think this is worth logging? This endpoint is always used at the same time as the assignment progress endpoint, in the frontend changes, and even if it was not I cannot see anything useful in knowing this was viewed. (We haven't previously logged when teachers just load an assignment but don't interact with it, and I also think the `VIEW_GROUPS_ASSIGNMENTS` event this may have been copied from isn't worth it either!)

Reply:
Fair enough! I was just copying the existing `VIEW_GROUPS_ASSIGNMENTS` log event as you said, but yeah I'll remove that one too 👍